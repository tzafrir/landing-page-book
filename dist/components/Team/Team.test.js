'use strict';

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _enzyme = require('enzyme');

var _enzymeAdapterReact = require('enzyme-adapter-react-16');

var _enzymeAdapterReact2 = _interopRequireDefault(_enzymeAdapterReact);

var _Team = require('./Team');

var _Team2 = _interopRequireDefault(_Team);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

(0, _enzyme.configure)({ adapter: new _enzymeAdapterReact2.default() });

describe('Team', function () {
  var props = void 0;
  var mounted = void 0;
  var team = function team() {
    if (!mounted) {
      mounted = (0, _enzyme.shallow)(_react2.default.createElement(_Team2.default, props));
    }
    return mounted;
  };
  describe('Team', function () {
    beforeEach(function () {
      props = {
        showBorder: true,
        showImage: true,
        showName: true,
        showJob: true,
        showSummary: true,
        imageCircle: true,
        imageBorder: false,
        socialGray: false,
        shadow: false,
        imageShadow: false,
        topColor: 'transparent',
        contentAlign: 'center',
        members: [{
          image: '/images/team/ninja.svg',
          job: 'Ninja',
          name: 'Leonardo',
          profile: [{
            social: 'facebook',
            url: 'https://www.facebook.com/JavaScriptTeam/'
          }, {
            social: 'twitter',
            url: 'https://twitter.com/front10hello?lang=es'
          }],
          summary: 'Leonardo, nickname Leo, is a fictional character and one of the four main characters in the Teenage Mutant Ninja Turtles comics and related media.'
        }, {
          image: '/images/team/bohemian.svg',
          job: 'Painter',
          name: 'Van Gogh',
          profile: [{
            social: 'facebook',
            url: 'https://www.facebook.com/JavaScriptTeam/'
          }, {
            social: 'email',
            url: 'https://front10.com'
          }]
        }]
      };
      mounted = undefined;
    });

    it('should render', function () {
      expect((0, _enzyme.shallow)(_react2.default.createElement(_Team2.default, props))).toMatchSnapshot();
    });

    it('always renders a div', function () {
      var divs = team().find('div');

      expect(divs.length).toBeGreaterThan(0);
    });
  });
});