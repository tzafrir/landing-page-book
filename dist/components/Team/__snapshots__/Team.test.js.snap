// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Team Team should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Team
    columnClassName="col-12 col-sm- col-md"
    contentAlign="center"
    imageBorder={false}
    imageCircle={true}
    imageShadow={false}
    members={
      Array [
        Object {
          "image": "/images/team/ninja.svg",
          "job": "Ninja",
          "name": "Leonardo",
          "profile": Array [
            Object {
              "social": "facebook",
              "url": "https://www.facebook.com/JavaScriptTeam/",
            },
            Object {
              "social": "twitter",
              "url": "https://twitter.com/front10hello?lang=es",
            },
          ],
          "summary": "Leonardo, nickname Leo, is a fictional character and one of the four main characters in the Teenage Mutant Ninja Turtles comics and related media.",
        },
        Object {
          "image": "/images/team/bohemian.svg",
          "job": "Painter",
          "name": "Van Gogh",
          "profile": Array [
            Object {
              "social": "facebook",
              "url": "https://www.facebook.com/JavaScriptTeam/",
            },
            Object {
              "social": "email",
              "url": "https://front10.com",
            },
          ],
        },
      ]
    }
    shadow={false}
    showBorder={true}
    showImage={true}
    showJob={true}
    showName={true}
    showSummary={true}
    socialGray={false}
    topColor="transparent"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="col-12 col-sm- col-md text-center mb-4 Team--center"
        >
          <div
            className="Team__Color"
          >
            <div
              className="Team__Color__Primary"
              style={
                Object {
                  "background": "transparent",
                }
              }
            />
          </div>
          <Card
            contentAlign="center"
            image="/images/team/ninja.svg"
            imageBorder={false}
            imageCircle={true}
            imageShadow={false}
            shadow={false}
            showBorder={true}
            subTitle="Ninja"
            summary="Leonardo, nickname Leo, is a fictional character and one of the four main characters in the Teenage Mutant Ninja Turtles comics and related media."
            summaryJustified={false}
            title="Leonardo"
          >
            <Social
              directLink={true}
              gray={false}
              rounded={true}
              type="facebook"
              url="https://www.facebook.com/JavaScriptTeam/"
            />
            <Social
              directLink={true}
              gray={false}
              rounded={true}
              type="twitter"
              url="https://twitter.com/front10hello?lang=es"
            />
          </Card>
        </div>,
        <div
          className="col-12 col-sm- col-md text-center mb-4 Team--center"
        >
          <div
            className="Team__Color"
          >
            <div
              className="Team__Color__Primary"
              style={
                Object {
                  "background": "transparent",
                }
              }
            />
          </div>
          <Card
            contentAlign="center"
            image="/images/team/bohemian.svg"
            imageBorder={false}
            imageCircle={true}
            imageShadow={false}
            shadow={false}
            showBorder={true}
            subTitle="Painter"
            summary=""
            summaryJustified={false}
            title="Van Gogh"
          >
            <Social
              directLink={true}
              gray={false}
              rounded={true}
              type="facebook"
              url="https://www.facebook.com/JavaScriptTeam/"
            />
            <Social
              directLink={true}
              gray={false}
              rounded={true}
              type="email"
              url="https://front10.com"
            />
          </Card>
        </div>,
      ],
      "className": "Team row",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "LeonardoNinja",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="Team__Color"
            >
              <div
                className="Team__Color__Primary"
                style={
                  Object {
                    "background": "transparent",
                  }
                }
              />
            </div>,
            <Card
              contentAlign="center"
              image="/images/team/ninja.svg"
              imageBorder={false}
              imageCircle={true}
              imageShadow={false}
              shadow={false}
              showBorder={true}
              subTitle="Ninja"
              summary="Leonardo, nickname Leo, is a fictional character and one of the four main characters in the Teenage Mutant Ninja Turtles comics and related media."
              summaryJustified={false}
              title="Leonardo"
            >
              <Social
                directLink={true}
                gray={false}
                rounded={true}
                type="facebook"
                url="https://www.facebook.com/JavaScriptTeam/"
              />
              <Social
                directLink={true}
                gray={false}
                rounded={true}
                type="twitter"
                url="https://twitter.com/front10hello?lang=es"
              />
            </Card>,
          ],
          "className": "col-12 col-sm- col-md text-center mb-4 Team--center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="Team__Color__Primary"
                style={
                  Object {
                    "background": "transparent",
                  }
                }
              />,
              "className": "Team__Color",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "Team__Color__Primary",
                "style": Object {
                  "background": "transparent",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Social
                  directLink={true}
                  gray={false}
                  rounded={true}
                  type="facebook"
                  url="https://www.facebook.com/JavaScriptTeam/"
                />,
                <Social
                  directLink={true}
                  gray={false}
                  rounded={true}
                  type="twitter"
                  url="https://twitter.com/front10hello?lang=es"
                />,
              ],
              "contentAlign": "center",
              "image": "/images/team/ninja.svg",
              "imageBorder": false,
              "imageCircle": true,
              "imageShadow": false,
              "shadow": false,
              "showBorder": true,
              "subTitle": "Ninja",
              "summary": "Leonardo, nickname Leo, is a fictional character and one of the four main characters in the Teenage Mutant Ninja Turtles comics and related media.",
              "summaryJustified": false,
              "title": "Leonardo",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "facebook",
                "nodeType": "class",
                "props": Object {
                  "directLink": true,
                  "gray": false,
                  "rounded": true,
                  "type": "facebook",
                  "url": "https://www.facebook.com/JavaScriptTeam/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "twitter",
                "nodeType": "class",
                "props": Object {
                  "directLink": true,
                  "gray": false,
                  "rounded": true,
                  "type": "twitter",
                  "url": "https://twitter.com/front10hello?lang=es",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "Van GoghPainter",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="Team__Color"
            >
              <div
                className="Team__Color__Primary"
                style={
                  Object {
                    "background": "transparent",
                  }
                }
              />
            </div>,
            <Card
              contentAlign="center"
              image="/images/team/bohemian.svg"
              imageBorder={false}
              imageCircle={true}
              imageShadow={false}
              shadow={false}
              showBorder={true}
              subTitle="Painter"
              summary=""
              summaryJustified={false}
              title="Van Gogh"
            >
              <Social
                directLink={true}
                gray={false}
                rounded={true}
                type="facebook"
                url="https://www.facebook.com/JavaScriptTeam/"
              />
              <Social
                directLink={true}
                gray={false}
                rounded={true}
                type="email"
                url="https://front10.com"
              />
            </Card>,
          ],
          "className": "col-12 col-sm- col-md text-center mb-4 Team--center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="Team__Color__Primary"
                style={
                  Object {
                    "background": "transparent",
                  }
                }
              />,
              "className": "Team__Color",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "Team__Color__Primary",
                "style": Object {
                  "background": "transparent",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Social
                  directLink={true}
                  gray={false}
                  rounded={true}
                  type="facebook"
                  url="https://www.facebook.com/JavaScriptTeam/"
                />,
                <Social
                  directLink={true}
                  gray={false}
                  rounded={true}
                  type="email"
                  url="https://front10.com"
                />,
              ],
              "contentAlign": "center",
              "image": "/images/team/bohemian.svg",
              "imageBorder": false,
              "imageCircle": true,
              "imageShadow": false,
              "shadow": false,
              "showBorder": true,
              "subTitle": "Painter",
              "summary": "",
              "summaryJustified": false,
              "title": "Van Gogh",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "facebook",
                "nodeType": "class",
                "props": Object {
                  "directLink": true,
                  "gray": false,
                  "rounded": true,
                  "type": "facebook",
                  "url": "https://www.facebook.com/JavaScriptTeam/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "email",
                "nodeType": "class",
                "props": Object {
                  "directLink": true,
                  "gray": false,
                  "rounded": true,
                  "type": "email",
                  "url": "https://front10.com",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col-12 col-sm- col-md text-center mb-4 Team--center"
          >
            <div
              className="Team__Color"
            >
              <div
                className="Team__Color__Primary"
                style={
                  Object {
                    "background": "transparent",
                  }
                }
              />
            </div>
            <Card
              contentAlign="center"
              image="/images/team/ninja.svg"
              imageBorder={false}
              imageCircle={true}
              imageShadow={false}
              shadow={false}
              showBorder={true}
              subTitle="Ninja"
              summary="Leonardo, nickname Leo, is a fictional character and one of the four main characters in the Teenage Mutant Ninja Turtles comics and related media."
              summaryJustified={false}
              title="Leonardo"
            >
              <Social
                directLink={true}
                gray={false}
                rounded={true}
                type="facebook"
                url="https://www.facebook.com/JavaScriptTeam/"
              />
              <Social
                directLink={true}
                gray={false}
                rounded={true}
                type="twitter"
                url="https://twitter.com/front10hello?lang=es"
              />
            </Card>
          </div>,
          <div
            className="col-12 col-sm- col-md text-center mb-4 Team--center"
          >
            <div
              className="Team__Color"
            >
              <div
                className="Team__Color__Primary"
                style={
                  Object {
                    "background": "transparent",
                  }
                }
              />
            </div>
            <Card
              contentAlign="center"
              image="/images/team/bohemian.svg"
              imageBorder={false}
              imageCircle={true}
              imageShadow={false}
              shadow={false}
              showBorder={true}
              subTitle="Painter"
              summary=""
              summaryJustified={false}
              title="Van Gogh"
            >
              <Social
                directLink={true}
                gray={false}
                rounded={true}
                type="facebook"
                url="https://www.facebook.com/JavaScriptTeam/"
              />
              <Social
                directLink={true}
                gray={false}
                rounded={true}
                type="email"
                url="https://front10.com"
              />
            </Card>
          </div>,
        ],
        "className": "Team row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "LeonardoNinja",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="Team__Color"
              >
                <div
                  className="Team__Color__Primary"
                  style={
                    Object {
                      "background": "transparent",
                    }
                  }
                />
              </div>,
              <Card
                contentAlign="center"
                image="/images/team/ninja.svg"
                imageBorder={false}
                imageCircle={true}
                imageShadow={false}
                shadow={false}
                showBorder={true}
                subTitle="Ninja"
                summary="Leonardo, nickname Leo, is a fictional character and one of the four main characters in the Teenage Mutant Ninja Turtles comics and related media."
                summaryJustified={false}
                title="Leonardo"
              >
                <Social
                  directLink={true}
                  gray={false}
                  rounded={true}
                  type="facebook"
                  url="https://www.facebook.com/JavaScriptTeam/"
                />
                <Social
                  directLink={true}
                  gray={false}
                  rounded={true}
                  type="twitter"
                  url="https://twitter.com/front10hello?lang=es"
                />
              </Card>,
            ],
            "className": "col-12 col-sm- col-md text-center mb-4 Team--center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="Team__Color__Primary"
                  style={
                    Object {
                      "background": "transparent",
                    }
                  }
                />,
                "className": "Team__Color",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "Team__Color__Primary",
                  "style": Object {
                    "background": "transparent",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Social
                    directLink={true}
                    gray={false}
                    rounded={true}
                    type="facebook"
                    url="https://www.facebook.com/JavaScriptTeam/"
                  />,
                  <Social
                    directLink={true}
                    gray={false}
                    rounded={true}
                    type="twitter"
                    url="https://twitter.com/front10hello?lang=es"
                  />,
                ],
                "contentAlign": "center",
                "image": "/images/team/ninja.svg",
                "imageBorder": false,
                "imageCircle": true,
                "imageShadow": false,
                "shadow": false,
                "showBorder": true,
                "subTitle": "Ninja",
                "summary": "Leonardo, nickname Leo, is a fictional character and one of the four main characters in the Teenage Mutant Ninja Turtles comics and related media.",
                "summaryJustified": false,
                "title": "Leonardo",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "facebook",
                  "nodeType": "class",
                  "props": Object {
                    "directLink": true,
                    "gray": false,
                    "rounded": true,
                    "type": "facebook",
                    "url": "https://www.facebook.com/JavaScriptTeam/",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "twitter",
                  "nodeType": "class",
                  "props": Object {
                    "directLink": true,
                    "gray": false,
                    "rounded": true,
                    "type": "twitter",
                    "url": "https://twitter.com/front10hello?lang=es",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "Van GoghPainter",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="Team__Color"
              >
                <div
                  className="Team__Color__Primary"
                  style={
                    Object {
                      "background": "transparent",
                    }
                  }
                />
              </div>,
              <Card
                contentAlign="center"
                image="/images/team/bohemian.svg"
                imageBorder={false}
                imageCircle={true}
                imageShadow={false}
                shadow={false}
                showBorder={true}
                subTitle="Painter"
                summary=""
                summaryJustified={false}
                title="Van Gogh"
              >
                <Social
                  directLink={true}
                  gray={false}
                  rounded={true}
                  type="facebook"
                  url="https://www.facebook.com/JavaScriptTeam/"
                />
                <Social
                  directLink={true}
                  gray={false}
                  rounded={true}
                  type="email"
                  url="https://front10.com"
                />
              </Card>,
            ],
            "className": "col-12 col-sm- col-md text-center mb-4 Team--center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="Team__Color__Primary"
                  style={
                    Object {
                      "background": "transparent",
                    }
                  }
                />,
                "className": "Team__Color",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "Team__Color__Primary",
                  "style": Object {
                    "background": "transparent",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Social
                    directLink={true}
                    gray={false}
                    rounded={true}
                    type="facebook"
                    url="https://www.facebook.com/JavaScriptTeam/"
                  />,
                  <Social
                    directLink={true}
                    gray={false}
                    rounded={true}
                    type="email"
                    url="https://front10.com"
                  />,
                ],
                "contentAlign": "center",
                "image": "/images/team/bohemian.svg",
                "imageBorder": false,
                "imageCircle": true,
                "imageShadow": false,
                "shadow": false,
                "showBorder": true,
                "subTitle": "Painter",
                "summary": "",
                "summaryJustified": false,
                "title": "Van Gogh",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "facebook",
                  "nodeType": "class",
                  "props": Object {
                    "directLink": true,
                    "gray": false,
                    "rounded": true,
                    "type": "facebook",
                    "url": "https://www.facebook.com/JavaScriptTeam/",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "email",
                  "nodeType": "class",
                  "props": Object {
                    "directLink": true,
                    "gray": false,
                    "rounded": true,
                    "type": "email",
                    "url": "https://front10.com",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
